modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/fragments/group/loading-validate/index.js

```diff
@@ -0,0 +1,236 @@
+/* eslint-disable @liferay/portal/no-global-fetch */
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const baseURL = Liferay.ThemeDisplay.getPortalURL();
+const userSessionId = Liferay.ThemeDisplay.getUserId();
+let userRoles = [];
+let userAdditionalInfosId = '';
+let sendingRole = false;
+const userAccountData = {
+	accountName: '',
+	userName: Liferay.ThemeDisplay.getUserName(),
+};
+
+const getUserAccountsById = async () => {
+	try {
+		const response = await fetch(
+			`${baseURL}/o/c/useradditionalinfos/?filter=r_userToUserAddInfo_userId eq '${userSessionId}'`,
+			{
+				headers: {
+					'content-type': 'application/json',
+					'x-csrf-token': Liferay.authToken,
+				},
+				method: 'GET',
+			}
+		);
+
+		if (response.ok) {
+			const userData = await response.json();
+			userData?.items?.map((data) => {
+				if (data.acceptInviteStatus === false) {
+					userAdditionalInfosId = data.id;
+					getMyUserAccount(data);
+					const arrayRoles = data.roles.split('/');
+					userRoles = arrayRoles.filter((role) => role !== '');
+				}
```

Comment:
Are you aware `getMyUserAccount` is a promise, being ruined in a loop in a synchronous way, right? is this expected?

Few possibilities here:

1. You could map over each userAditionalinfos using (for of) 
2. Do a Promise All

```suggestion
			const userAdditionalInfos = await response.json();
			
			userAdditionalInfos?.items?.map((userAdditionalInfos) => {
				if (userAdditionalInfo.acceptInviteStatus === false) {
					userAdditionalInfosId = userAdditionalInfos.id;
					
					getMyUserAccount(userAdditionalInfos);
					
					const arrayRoles = userAdditionalInfos.roles.split('/');
					
					userRoles = arrayRoles.filter((role) => role !== '');
				}
```